crates/red_knot_python_semantic/src/types.rs

```diff
@@ -162,6 +164,8 @@ fn bindings_ty<'db>(
     }
 }
 
+type DeclaredTypeResult<'db> = Result<Type<'db>, (Type<'db>, Box<[Type<'db>]>)>;
```

Comment:
I thought you don't like `Result` :laughing:

Reply:
I changed my mind -- I discussed this in a comment above.